packages/web-runtime/src/services/uppyService.ts

```diff
@@ -0,0 +1,223 @@
+      retryDelays: [0, 3000, 5000, 10000, 20000]
+    }
+
+    const tusPlugin = this.uppy.getPlugin('Tus')
+    if (tusPlugin) {
+      tusPlugin.setOptions(tusPluginOptions)
+      return
+    }
+
+    this.uppy.use(CustomTus, tusPluginOptions)
+  }
+
+  useXhr({ headers }: { headers: { [key: string]: string } }) {
+    const xhrPluginOptions: XHRUploadOptions = {
+      endpoint: '',
+      method: 'put',
+      headers,
+      getResponseData() {
+        return {}
+      }
+    }
+
+    const tusPlugin = this.uppy.getPlugin('XHRUpload')
+    if (tusPlugin) {
+      tusPlugin.setOptions(xhrPluginOptions)
+      return
+    }
+
+    this.uppy.use(XHRUpload, xhrPluginOptions)
+  }
+
+  useDropTarget({
+    targetSelector,
+    uppyService
+  }: {
+    targetSelector: string
+    uppyService: UppyService
+  }) {
+    if (this.uppy.getPlugin('DropTarget')) {
+      return
+    }
+    this.uppy.use(CustomDropTarget, {
+      target: targetSelector,
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      uppyService
+    })
+  }
+
+  useStatusBar({
+    targetSelector,
+    getText
+  }: {
+    targetSelector: string
+    getText: (msgid: string) => string
+  }) {
+    if (this.uppy.getPlugin('StatusBar')) {
+      return
+    }
+
+    this.uppy.use(StatusBar, {
+      id: 'StatusBar',
+      target: targetSelector,
+      hideAfterFinish: true,
+      showProgressDetails: true,
+      hideUploadButton: false,
+      hideRetryButton: false,
+      hidePauseResumeButton: false,
+      hideCancelButton: false,
+      doneButtonHandler: null,
+      locale: {
+        strings: {
+          uploading: getText('Uploading'),
+          complete: getText('Complete'),
+          uploadFailed: getText('Upload failed'),
+          paused: getText('Paused'),
+          retry: getText('Retry'),
+          cancel: getText('Cancel'),
+          pause: getText('Pause'),
+          resume: getText('Resume'),
+          done: getText('Done'),
+          filesUploadedOfTotal: {
+            0: getText('%{complete} of %{smart_count} file uploaded'),
+            1: getText('%{complete} of %{smart_count} files uploaded')
+          },
+          dataUploadedOfTotal: getText('%{complete} of %{total}'),
+          xTimeLeft: getText('%{time} left'),
+          uploadXFiles: {
+            0: getText('Upload %{smart_count} file'),
+            1: getText('Upload %{smart_count} files')
+          },
+          uploadXNewFiles: {
+            0: getText('Upload +%{smart_count} file'),
+            1: getText('Upload +%{smart_count} files')
+          },
+          upload: getText('Upload'),
+          retryUpload: getText('Retry upload'),
+          xMoreFilesAdded: {
+            0: getText('%{smart_count} more file added'),
+            1: getText('%{smart_count} more files added')
+          },
+          showErrorDetails: getText('Show error details')
+        }
+      }
+    })
+  }
```

Comment:
We need to make sure this actually works, Pascal mentioned gettext in ts files might be problematic